nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -245,7 +268,14 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     curvature = 2.0 * carrot_pose.pose.position.y / carrot_dist2;
   }
 
-  linear_vel = desired_linear_vel_;
+  // Setting the velocity direction
+  double sign = 1.0;
+
+  if(reverse_driving_){
+    sign = carrot_pose.pose.position.x >= 0.0 ? 1.0:-1.0;
```

Comment:
These semantics would make a whole lot more sense if you just had a flag you could work with that it detected that it should change signs due to a cusp

Reply:
I just mean in terms of this code, you end up finding the "sign" in a couple of places that might be better served with a boolean that is just passed to a function to use instead so its not being re-found in multiple places based on derived values